src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/DirectoryServerCollection.cs

```diff
@@ -14,7 +15,7 @@ public class DirectoryServerCollection : CollectionBase
         internal readonly DirectoryContext context;
         internal bool initialized;
         internal readonly Hashtable? changeList;
```

Comment:
Dictionary?

Reply:
The code is explicitly synchronizing access to this internal member.  It'd require review of all uses to ensure that was safe.  In general, replacing Hashtable with Dictionary needs review as they have subtly different thread-safety guarantees, and code frequently relies on Hashtable's, which enables lock-free reads with a single concurrent writer.